src/utils/remote.c

```diff
@@ -238,10 +238,67 @@ bool launch_agent(void)
 		fio_redirect(infd[0], outfd[1], errfd[0]); /* write to stdout */
 	}
 
-	/* Make sure that remote agent has the same version
-	 * TODO: we must also check PG version and fork edition
-	 */
-	agent_version = fio_get_agent_version();
+
+	/* Make sure that remote agent has the same version, fork and other features to be binary compatible */
+	{
+		char payload_buf[1024];
+		fio_get_agent_version(&agent_version, payload_buf, sizeof payload_buf);
+		check_remote_agent_compatibility(agent_version, payload_buf, sizeof payload_buf);
+	}
+
+	return true;
+}
+
+#define COMPATIBILITY_VAL_STR(macro) #macro, macro
+#define COMPATIBILITY_VAL_INT_HELPER(macro, helper_buf, buf_size) (snprintf(helper_buf, buf_size, "%d", macro), helper_buf)
+#define COMPATIBILITY_VAL_INT(macro, helper_buf, buf_size) #macro, COMPATIBILITY_VAL_INT_HELPER(macro, helper_buf, buf_size)
+
+#define COMPATIBILITY_VAL_SEPARATOR "="
+#define COMPATIBILITY_LINE_SEPARATOR "\n"
+
+/*
+ * Compose compatibility string to be sent by pg_probackup agent
+ * through ssh and to be verified by pg_probackup peer.
+ * Compatibility string contains postgres essential vars as strings
+ * in format "var_name" + COMPATIBILITY_VAL_SEPARATOR + "var_value" + COMPATIBILITY_LINE_SEPARATOR
+ */
+size_t prepare_compatibility_str(char* compatibility_buf, size_t compatibility_buf_size)
+{
+	char compatibility_val_int_macro_helper_buf[32];
+	char* compatibility_params[] = {
+		COMPATIBILITY_VAL_STR(PG_MAJORVERSION),
+#ifdef PGPRO_EDN
+		//TODO REVIEW can use edition.h/extract_pgpro_edition() or similar
+		COMPATIBILITY_VAL_STR(PGPRO_EDN),
+#endif
+		//TODO REVIEW remove? no difference between 32/64 in global/pg_control.
```

Comment:
There's no difference between 32/64 in global/pg_control. do we still make it core compatibility param?